service/controller.go

```diff
@@ -2005,6 +2009,12 @@ func (s *service) ListVolumes(
 			maxEntries = int(req.MaxEntries)
 		)
 
+		// To prevent DOS in the gRPC server, enforcing list volumes pagination
+		// with the limit of MaxVolumeListEntries entries per request.
+		if maxEntries == 0 || maxEntries > MaxVolumeListEntries {
+			maxEntries = MaxVolumeListEntries
```

Comment:
should we log a message saying maxEntries is bring set to MaxVolumeListEntries here? 
It's an edge case but might be confusing if someone checks the driver logs and sees that the request max entries field is being modified by the driver

Reply:
I intentionally didn''t put the log here, since it's actually not an edge case. So far all request I've seen had MaxEntries=0, so I didn't want to contaminate the log with this essentially unconditional message. Do you think we still need it?

Reply:
@alexemc - How does this impact CSI compliance?

Reply:
This is explained in the PR description.